### ISSUE TITLE
0.6.3 update console output issue

### ISSUE THREAD

**Issue by nechtmarrie** [Pterodactyl/Panel#522]
After installing the 0.6.3 panel update the console for servers will read out the logging of the past +-3 houres 1 line at a time and appending it to the bottom of the console output.
To see actual new command feedback/console logging you need to wait about 10 minuted for the log to finish reading.

**Comment by DaneEveritt** [comment]
That sounds like you have a very, very low throttle limit set, or an insane amount of data is being returned from the daemon, which shouldn't be the case.

/cc @Pterodactyl/support are you able to replicate this at all?

**Comment by kaganus** [comment]
a lot of users have this, including me. whenever i navigate into the console page of any server, logs from 10-15 minutes ago start to spam it until it catches up. asked about this in #support channel of discord a few times too, no one really replied

**Comment by DaneEveritt** [comment]
Can you do me a favor and see if modifying `CONSOLE_PUSH_COUNT` and `CONSOLE_PUSH_FREQ` has any effect on the issue you're seeing? Those default to `10` and `200` respectively, which means every 200ms 10 new lines are flushed to the console. Try changing those to push 100 lines every 100ms, or quicker depending on what you think.

You'll need to re-run `php artisan config:cache` each time you edit them, and they can be defined in the `.env` file.

Let me know if that makes any difference.

**Comment by nechtmarrie** [comment]
@DaneEveritt I had these values set to the following before changing them to the values you specified.
CONSOLE_PUSH_FREQ=250
CONSOLE_PUSH_COUNT=10

Changed them to 
CONSOLE_PUSH_FREQ=200
CONSOLE_PUSH_COUNT=10

Still the same result i'm afraid.

**Comment by schrej** [comment]
I would just force print the initial log by replacing lines [126](https://github.com/Pterodactyl/Panel/blob/develop/public/themes/pterodactyl/js/frontend/console.js#L126)-128 in `/public/themes/pterodactyl/js/frontend/console.js` with
```js
data.split(/\n/g).forEach(function (item) {
  $terminal.append(
    '<div class="cmd">' + AnsiUp.ansi_to_html(item + '\u001b[0m') + '</div>'
  );
});
window.scrollToBottom();
```
Or better, put that stuff into a function and call it at both spots that need it to avoid duplicated code.
That console thing needs a rewrite after that hotfix anyways though.

**Comment by nechtmarrie** [comment]
@schrej That fixed the issue for now :)

**Comment by Polarcraft** [comment]
Dane is going through and fixing issues with the console, expect an update for it soon.

**Comment by DaneEveritt** [comment]
closed in 7463bea

**Comment by DaneEveritt** [comment]
![](https://zippy.gfycat.com/EasyGenuineBittern.gif)
